NIPTool/crud/find.py

```diff
@@ -0,0 +1,57 @@
+from typing import Iterable, List
+
+from NIPTool.adapter import NiptAdapter
+from NIPTool.models.database import Batch, User, Sample
+
+
+def user(adapter: NiptAdapter, user_name: str) -> User:
+    """Find user from user collection"""
+    raw_user: dict = adapter.user_collection.find_one({"_id": user_name})
+    return User(**raw_user)
+
+
+def sample_to_model(raw_sample: dict) -> Sample:
+    raw_sample["SampleID"] = raw_sample.pop("_id")
+    return Sample(**raw_sample)
+
+
+def sample(adapter: NiptAdapter, sample_id: str) -> Sample:
+    """Find one sample from the sample collection"""
+
+    return adapter.sample_collection.find_one({"_id": sample_id})
+
+
+def batch_to_model(raw_batch: dict) -> Batch:
+    """Convert a database dict to batch object"""
+    raw_batch["batch_id"] = raw_batch.pop("_id")
+    return Batch(**raw_batch)
+
+
+def batch(adapter: NiptAdapter, batch_id: str) -> Batch:
+    """Find one batch from the batch collection"""
+    raw_batch: dict = adapter.batch_collection.find_one({"_id": batch_id})
+    return batch_to_model(raw_batch)
+
+
+def batches(adapter: NiptAdapter) -> List[Batch]:
+    """Return all batches from the batch collection"""
+    raw_batches: Iterable[dict] = adapter.batch_collection.find()
+    return [batch_to_model(raw_batch) for raw_batch in raw_batches]
+
+
```

Comment:
Aggragations do not return lists of Sample or Batch objects. The format of the content is totally depending on the pipe that is inserted

Reply:
Yes I suspected that. I suggest that we implement functions for the different types of aggregations. Do you think that would be possible ?